syntax = "proto3";

import "google/protobuf/any.proto";

service Apns {
    rpc SetUserApns(SetUserApnsRequest) returns (DefaultResponse);
    rpc SendNotificationToUser(SendNotificationRequest) returns (DefaultResponse);
}

// 设置用户Apns请求的参数
message SetUserApnsRequest {
    string sid = 1; // 用户学号
    string apn = 2; // 用户apn
}

// 设置用户Apns请求的返回值
message DefaultResponse {
    int32 status = 1;   // 请求结果，成功则返回1，失败返回0
    string msg = 2; // 请求结果相关信息，报错时打印错误信息，成功时返回success
    optional google.protobuf.Any data = 3;
}

message SendNotificationRequest {
    message AppleNotification {
        message AppleAlert {
            string title = 1;   // 通知标题
            optional string subtitle = 2;   // 通知标题，为空字符串则不发送该字段
            string body = 3;   // 通知标题
        }

        AppleAlert alert = 1;   // Apple Remote Notification
        int32 badge = 2;    // 应用程序图标上徽章中显示的数字。指定0以删除当前徽章（如果有的话），置-1以不发送该字段
        string category = 3;    // 通知的类型，为空字符串则不发送该字段
    }

    string sid = 1; // 用户学号
    AppleNotification notification = 2;     // 通知信息设置
}